src/Symfony/Component/Security/Http/AccessToken/Oidc/OidcTokenHandler.php

```diff
@@ -43,9 +41,12 @@ public function __construct(
         private string $audience,
         private array $issuers,
         private string $claim = 'sub',
-        private ?LoggerInterface $logger = null,
-        private ClockInterface $clock = new Clock()
+        private ClockInterface|LoggerInterface|null $clock = new Clock()
     ) {
+        if (!$clock || $clock instanceof LoggerInterface) {
+            $this->clock = new Clock();
```

Comment:
You also need to account for the case where the clock is pass explicitly in the old position instead of ignoring it and assuming that the default clock is used.